/* Copyright 2022 The MediaPipe Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package mediapipe.tasks;

import "mediapipe/framework/calculator.proto";
import "mediapipe/tasks/cc/components/segmenter_options.proto";
import "mediapipe/util/label_map.proto";

message TensorsToSegmentationCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional TensorsToSegmentationCalculatorOptions ext = 458105876;
  }

  optional SegmenterOptions segmenter_options = 1;

  // Identifying information for each classification label.
  map<int64, mediapipe.LabelMapItem> label_items = 2;
}
